src/Microsoft.AspNet.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -171,5 +172,51 @@ protected virtual JsonSerializer CreateJsonSerializer()
         {
             return JsonSerializer.Create(SerializerSettings);
         }
+
+        private ModelMetadata GetPathMetadata(ModelMetadata metadata, string path)
+        {
+            var index = 0;
+            while (index >= 0 && index < path.Length)
```

Comment:
Not in a path we understand. However we can't guarantee Json.NET will never provide a path such as `"property["`.